5.5/run-mysqld.sh

```diff
@@ -60,10 +57,13 @@ function wait_for_mysql {
 	done
 }
 
-if [ ! -d '/var/lib/mysql/mysql' ]; then
+if [ "$1" = "mysqld" -a ! -d '/var/lib/mysql/data/mysql' ]; then
+
+	shift
+	check_env_vars
 
 	echo 'Running mysql_install_db'
-	mysql_install_db --datadir=/var/lib/mysql
+	mysql_install_db --datadir=/var/lib/mysql/data
```

Comment:
i hate having hardcoded paths in scripts as they are harder to maintain... make "/var/lib/mysql" variable